packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,308 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { DevicePermission } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Interact with image and video. It lets the app developer ask the user to get images or video from their camera / camera roll / file system.
+ * @beta
+ */
+export namespace visualMedia {
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * All properties common to Image and Video Props
+   * @beta
+   */
+  interface VisualMediaProps {
+    /**
+     * max limit of media allowed to be selected in one go, current max limit is 10.
+     */
+    maxVisualMediaCount: number;
+  }
+
+  /**
+   * The required value of the visualMedia files from gallery
+   * @beta
+   */
+  export interface GalleryProps {
+    /**
+     * @hidden
+     * The visualMedia source
+     */
+    source: Source.Gallery;
+  }
+  /**
+   * The required value of the visualMedia files from camera
+   * @beta
+   */
+  export interface CameraProps {
+    /**
+     * @hidden
+     * The visualMedia source
+     */
+    source: Source.Camera;
+    /**
+     * Optional; indicate if user is allowed to move between front and back camera
+     * Default value is FrontOrRear
+     */
+    cameraRestriction?: CameraRestriction;
+  }
+
+  /**
+   * Indicate if user is allowed to move between front and back camera or stay in front/back camera only
+   * If the camera option requested by the app isn't available, the SDK will silently default to the platform's standard camera.
+   * @beta
+   */
+  export enum CameraRestriction {
+    /** User can move between front and back camera */
+    FrontOrRear = 1,
+    /** User can only use the front camera */
+    FrontOnly = 2,
+    /** User can only use the back camera */
+    RearOnly = 3,
+  }
+  /**
+   * Specifies the image source
+   * @beta
+   */
+  export enum Source {
+    /** visual media source is camera. */
+    Camera = 1,
+    /** visual media source is gallery. */
+    Gallery = 2,
+  }
+
+  /**
+   * VisualMediaFile object that can be used to represent image or video
+   *
+   * @beta
+   */
+  export class VisualMediaFile {
+    /**
+     * @hidden
+     * This is UUID for hub-sdk to identify the image
+     */
+    public id: string;
+    /**
+     * This is the base64 content of file.
+     * If app needs to use this directly in HTML tags, it should convert this to a data url.
+     */
+    public content: string;
+    /**
+     * The size of file represented in VisualMediaFile in KB
+     */
+    public sizeInKB: number;
+
+    /**
+     * Name of the file (not includs the extension)
+     */
+    public name: string;
+
+    /**
+     * file's MIME type.
+     */
+    public mimeType: string;
```

Comment:
Are we going to support all the [mimeTypes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types) available?

Reply:
Based on the lens-sdk, which is the library that the host apps are currently using, and also based on what the current media API supports, we have no limitations on the mimeTypes. I updated the comments with the link you mentioned, to make sure app developer get to know more information about the mimeTypes.

Reply:
Can you point me to the link where I can find lens-sdk mimeTypes?

Reply:
> Can you point me to the link where I can find lens-sdk mimeTypes?

I totally understand your point. In the clipboard API, there is the ClipboardSupportedMimeType to restrict the mimeType. Using the ClipboardSupportedMimeType enum class is much better than using the string type directly. However, to the best of my knowledge, the Lens-sdk has no limitations on the mimeTypes, and that's why I'm keeping the string type from the current media API. 
@Ella-ly, please correct me if I'm wrong. It would be great if you could update us on anything related to mimeType from the lens-SDK side. Thanks!

Reply:
Synced offline with @shrshindeMSFT , since we are lacking documentation on the lens-sdk mimeType, I updated with this public link: https://docs.lens.xyz/docs/metadata-standards#supported-mime-types-for-imagesaudiovideos